### PR TITLE
History and Playlist

### PR DESCRIPTION
**Pull request by AbhishekkGautam**
### History page 
- videos card
- whenever the user clicks on any video, that video will add to the history. 
- user can remove a particular video from history.
- user can also clear all history.

### Playlist management.
- user can create a playlist, and add/remove any video by just clicking on the `Save to playlist` option on the video card.
- user can view all the created playlists by navigating to `/playlist`.
- user can also create & delete a particular playlist on the playlist page itself.
- user can view a list of videos in a particular playlist.
- user can also delete any video from a particular playlist.

### Create new playlist & add/remove a particular video

Screen Shot 2022-04-04 at 12 22 43 AM

### View all playlists.

Screen Shot 2022-04-04 at 12 26 32 AM

### delete a particular video from playlist

Screen Shot 2022-04-04 at 12 25 45 AM

### CODE REVIEW

**Comment by pritam-kr** [comment]
- Code looks clean 
- I have some points to write 
    - netlify, after refreshing pages jumps to a 404 page 
    - Add a pre thumbnail if the playlist has no video
- UI and UX 🔥

**Review by pritam-kr — APPROVED** [review]

File: src/components/AddPlaylistModal/AddPlaylistModal.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+                  name="title"
+                  placeholder="Enter playlist name"
+                  value={title || ""}
+                  onChange={e => setTitle(e.target.value)}
+                />
+                <div className="btn-container">
+                  <button
+                    className="form-btn"
+                    onClick={() => {
+                      createPlaylistService(title, token, dispatch);
```
Comment (msdeshmukh009) [inline src/components/AddPlaylistModal/AddPlaylistModal.js:53]: Maybe you should not do dispatch in the service function.

**Review by msdeshmukh009 — APPROVED** [review]
The rest is looking good. UI looks really nice.

File: src/components/AddPlaylistModal/AddPlaylistModal.js (hunk 1 above)
Comment (AbhishekkGautam) [reply src/components/AddPlaylistModal/AddPlaylistModal.js:53]: Ooh. The dispatch function will get the data/error from api & save it in respective states. Then I'll do the rest calculations using context data.

**Pull request merged**